torch/export/_remove_auto_functionalized_pass.py

```diff
@@ -80,14 +60,29 @@ def unsafe_remove_auto_functionalized_pass(
                 if user.args[1] == 0:
                     user.replace_all_uses_with(new_node)
 
-                    # Same case as above, update the output spec if getitem result used in an output node
-                    for spec in output_specs:
-                        if spec.arg.name == user.name:
-                            spec.arg.name = new_node.name
-                            break
-
         new_node.meta["val"] = node.meta["val"][: len(func._schema.returns)]
         ep.graph.erase_node(node)
 
     ep.graph.eliminate_dead_code()
+
+
+def unsafe_remove_auto_functionalized_pass(
+    ep: ExportedProgram,
+) -> ExportedProgram:
+    """
+    This pass removes an instances of the higher order op 'auto_functionalized',
+    and modifies the calling EP inplace to have the original mutator op.
+    This pass doesn't perform safety checks to make sure that this inplace mutation is safe.
+    """
+    auto_functionalize_nodes: List[torch.fx.Node] = []
+    for module in ep.graph_module.modules():
+        if not isinstance(module, torch.fx.GraphModule):
+            continue
+        for node in ep.graph.nodes:
+            if node.op == "call_function" and node.target is auto_functionalized:
+                auto_functionalize_nodes.append(node)
+
+    with ep.graph_module._set_replace_hook(ep.graph_signature.get_replace_hook()):
```

Comment:
cc @JacobSzwejbka -- I also modified the remove_auto_functionalize pass to use the set_replace_hooks which automatically updates the graph signature (forgot this handy context manager existed oops).